src/SprykerMiddleware/Zed/Process/Business/ProcessBusinessFactory.php

```diff
@@ -2,50 +2,212 @@
 use SprykerMiddleware\Zed\Process\Business\Translator\TranslatorFunction\TranslatorFunctionResolver;
 use SprykerMiddleware\Zed\Process\Business\Translator\TranslatorInterface;
 use SprykerMiddleware\Zed\Process\Business\Writer\WriterInterface;
 use SprykerMiddleware\Zed\Process\Business\Writer\WriterResolver;
+use SprykerMiddleware\Zed\Process\ProcessDependencyProvider;
 
 /**
  * @method \SprykerMiddleware\Zed\Process\ProcessConfig getConfig()
  */
 class ProcessBusinessFactory extends AbstractBusinessFactory
 {
+    use LoggerTrait;
+
+    /**
+     * @param \Generated\Shared\Transfer\ProcessSettingsTransfer $processSettingsTransfer
+     *
+     * @return \SprykerMiddleware\Zed\Process\Business\Process\ProcessorInterface
+     */
+    public function createProcessor(ProcessSettingsTransfer $processSettingsTransfer): ProcessorInterface
+    {
+
+        $stagesPluginsList = $this->getStagePluginsListForProcess($processSettingsTransfer->getName());
+        $logger = $this->getLogger($this->getProcessLoggerConfig($processSettingsTransfer));
+        return new Processor(
+            $this->getProcessIterator($processSettingsTransfer),
+            $this->createPipeline($stagesPluginsList, $logger),
+            $this->getPreProcessHookStack($processSettingsTransfer->getName()),
+            $this->getPostProcessHookStack($processSettingsTransfer->getName()),
+            $logger
+        );
+    }
+
+    /**
+     * @param string $processName
+     *
+     * @return array
+     */
+    protected function getStagePluginsListForProcess(string $processName): array
+    {
+        $stages = $this->getProvidedDependency(ProcessDependencyProvider::MIDDLEWARE_PROCESS_STAGES);
+        return $stages[$processName];
+    }
+
+    /**
+     * @param string $processName
+     *
+     * @return array
+     */
+    protected function getPreProcessHookStack(string $processName): array
+    {
+        $preProcessHooks = $this->getProvidedDependency(ProcessDependencyProvider::MIDDLEWARE_PRE_PROCESSOR_HOOKS_STACK);
+        return $preProcessHooks[$processName];
+    }
+
+    /**
+     * @param string $processName
+     *
+     * @return array
+     */
+    protected function getPostProcessHookStack(string $processName): array
+    {
+        $postProcessHooks = $this->getProvidedDependency(ProcessDependencyProvider::MIDDLEWARE_POST_PROCESSOR_HOOKS_STACK);
+        return $postProcessHooks[$processName]?:[];
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProcessSettingsTransfer $processSettingsTransfer
+     *
+     * @return \Iterator
+     */
+    protected function getProcessIterator(ProcessSettingsTransfer $processSettingsTransfer): Iterator
+    {
+        $iterators = $this->getProcessIteratorsList();
+        return $iterators[$processSettingsTransfer->getName()]($processSettingsTransfer->getIteratorSettings());
+    }
+
+    /**
+     * @return array
+     */
+    protected function getProcessIteratorsList(): array
```

Comment:
This should be configured in the `DependencyProvider`!

Reply:
Should be changed in future